packages/manager/cypress/e2e/core/cloudpulse/widget-level-dimension-filter.spec.ts

```diff
@@ -0,0 +1,586 @@
+const ADD_FILTER_TEXT = 'Add Filter';
+const APPLY_LABEL = 'Apply';
+const CANCEL_LABEL = 'Cancel';
+const ENTER_VALUE_PLACEHOLDER = 'Enter a Value';
+const DIMENSION_FILTER_BADGE =
+  '[data-qa-badge="dimension-filter-badge-content"]';
+const DIMENSION_FILTER_PREFIX = '[data-qa-dimension-filter="dimension_filters.';
+const DIMENSION_FIELD_SUFFIX = '-dimension-field"]';
+const GET_METRICS_ALIAS = '@getMetrics';
+const METRICS_REQUESTS_COUNT = 6;
+const FILTER_BTN_ALIAS = 'filterBtn';
+const CONTAIN_TEXT = 'contain.text';
+const SELECT_UP_TO_5_TEXT = 'Select up to 5 filters.';
+const CLEAR_ICON_TESTID = 'clear-icon';
+
+// Helper functions
+const openFilterDrawer = (widgetIndex: number = 0) => {
+  ui.button
+    .findByAttribute(
+      ARIA_LABEL,
+      `Widget Dimension Filter${dashboard.widgets[widgetIndex].label}`
+    )
+    .should(VISIBLE)
+    .and(NOT_DISABLED)
+    .as(FILTER_BTN_ALIAS);
+
+  cy.get(`@${FILTER_BTN_ALIAS}`).should(ENABLED).click();
+  ui.drawer.find().should('exist');
+};
+
+const addFilter = ({ index, dimension, operator, value }: AddFilterProps) => {
+  ui.button.findByTitle(ADD_FILTER_TEXT).click();
+  cy.get(`[data-testid="dimension_filters.${index}-id"]`).within(() => {
+    ui.autocomplete.findByLabel('Dimension').should(VISIBLE).type(dimension);
+
+    ui.autocompletePopper.findByTitle(dimension).should(VISIBLE).click();
+
+    ui.autocomplete.findByLabel('Operator').should(VISIBLE).type(operator);
+
+    ui.autocompletePopper.findByTitle(operator).should(VISIBLE).click();
+
+    cy.get(`[data-qa-dimension-filter="dimension_filters.${index}-value"]`)
+      .findByPlaceholderText(ENTER_VALUE_PLACEHOLDER)
+      .should(VISIBLE)
+      .type(value);
+  });
+};
+
+const verifyDrawerContent = (widgetIndex: number = 0) => {
+  ui.drawer.find().within(() => {
+    cy.get('[data-testid="drawer-title"]')
+      .should(VISIBLE)
+      .and(CONTAIN_TEXT, DIMENSION_FILTERS_TEXT);
+
+    cy.get('[data-qa-id="filter-drawer-subtitle"]')
+      .should(VISIBLE)
+      .and(CONTAIN_TEXT, dashboard.widgets[widgetIndex].label);
+
+    cy.get('[data-qa-id="filter-drawer-selection-title"]')
+      .should(VISIBLE)
+      .and(CONTAIN_TEXT, SELECT_UP_TO_5_TEXT);
+  });
+};
+
+const verifyBadgeCount = (count: string) => {
+  cy.get(`@${FILTER_BTN_ALIAS}`).within(() => {
+    cy.get(DIMENSION_FILTER_BADGE).should(VISIBLE).and(CONTAIN_TEXT, count);
+  });
+};
+
+const verifyDimensionsExcludeFilters = (excludedFilters: string[]) => {
+  ui.autocompletePopper
+    .find()
+    .should(VISIBLE)
+    .find('[data-qa-option]')
+    .then(($options) => {
+      const dimensionOptions = $options
+        .toArray()
+        .map((el) => el.textContent?.trim() || '')
+        .filter((text) => text !== '');
+
+      // Verify each excluded filter (global filter) is NOT present in dimension options array
+      excludedFilters.forEach((excludedFilter) => {
+        expect(dimensionOptions).to.not.include(excludedFilter);
+      });
+    });
```

Comment:
@shnagend , lets not use it...we need to find an alternative way, can we use the below and see if it works

```suggestion
ui.autocompletePopper.find().within(() => {
    excludedFilters.forEach((filter) => {
      cy.findByText(filter).should('not.exist');
    });
  });
```